### PR TITLE
entc/gen/template: add GoLand annotation for template data-types

### PR DESCRIPTION
**Pull request by a8m**
🤯 🤯 🤯 🤯 🤯 🤯 🤯 🤯 🤯 🤯 🤯 🤯 🤯 🤯 
This annotation enables the auto-completion of templates in GoLand.  

image

### CODE REVIEW

**Comment by a8m** [comment]
cc @alexsn

File: entc/gen/template/predicate.tmpl [hunk 1]
```diff
@@ -17,7 +19,7 @@ import (
 	"entgo.io/ent/dialect/sql"
 )
 
-{{ range $_, $n := $.Nodes -}}
+{{ range $n := $.Nodes -}}
```
Comment (giautm) [inline entc/gen/template/predicate.tmpl:22]: hey @a8m, I believe this fix is wrong. See: https://play.golang.org/p/TXAOkWN43pp

Screen Shot 2021-04-09 at 17 57 47

Comment (a8m) [reply entc/gen/template/predicate.tmpl:22]: The range clause works a bit different in templates (see the `text/template` [doc](https://golang.org/pkg/text/template/#hdr-Variables)):

> Note that if there is only one variable, it is assigned the element; this is opposite to the convention in Go range clauses.

But I still appreciate the review. Thanks 🙏

Comment (giautm) [reply entc/gen/template/predicate.tmpl:22]: Yeah, I have checked the doc but missed the note. :)))

**Pull request merged**